sematic/resolvers/local_resolver.py

```diff
@@ -441,6 +467,23 @@ def _save_graph(self):
         self._buffer_artifacts.clear()
         self._buffer_edges.clear()
```

Comment:
Why here? 🤔

Reply:
because the "happy path" disconnect doesn't get called if the resolver errors out. If we want to make sure that the disconnect always happens, we will need it here. Really it would be ideal if we had some `resolution_did_terminate` that ALWAYS gets called regardless of which terminal status is reached. Then we can put cleanup stuff there.